public/pages/CreateDetector/components/ConfigureFieldMapping/containers/ConfigureFieldMapping.tsx

```diff
@@ -269,9 +269,13 @@ export default class ConfigureFieldMapping extends Component<
         const unmappedRuleFields = new Set(mappingsView.response.unmapped_field_aliases);
 
         Object.keys(mappingsView.response.properties).forEach((ruleFieldName) => {
-          // Filter the mappings view to include only the rule fields for the enabled rules
-          if (!ruleFieldsForEnabledRules.has(ruleFieldName)) {
+          // Filter the mappings view to include only the rule fields for the enabled rules except for the timestamp field
+          if (
+            ruleFieldsForEnabledRules.size === 0 ||
+            (!ruleFieldsForEnabledRules.has(ruleFieldName) && ruleFieldName !== 'timestamp')
```

Comment:
Nitpick; not blocking: Would it be slightly more efficient to swap the checks on this line to the following?
```
(ruleFieldName !== 'timestamp' && !ruleFieldsForEnabledRules.has(ruleFieldName))